### PR TITLE
fix: Tag image after pushing to sandbox internal registry

### LINKED ISSUE #75: Tag image with internal-registry-host/namespace/image-name in local registry afte pushing image to sandbox cluster
This later would allow use the image to create local container and deploy it to developer sandbox cluster.

### PR DESCRIPTION
**Pull request by jeffmaury**
Local tag thould match destination tag uses to push to internal registry.

Fix #75.

1. Login to Sandbox
2. Select one of your images and run Push to Red Hat Developer Sandbox
3. Select the new tag for the image with the Sandbox URL, start it
4. Once the container is up, select the Deploy to Kubernetes and once the pod is created, click on the link to the OpenShift console to verify pod is started

sandbox-push

### CODE REVIEW

**Review by dgolovin — DISMISSED** [review]
LGTM

File: src/extension.ts [hunk 1]
```diff
@@ -125,39 +124,47 @@ export async function pushImageToOpenShiftRegistry(image: ImageInfo): Promise<vo
...
+          );
+        } else {
+          await extensionApi.window.showInformationMessage(
+            `The image '${image.name}:${image.tag}' has been successfully pushed to to Developer Sandbox cluster. '${targetSb}'.`,
+          );
+        }
       } catch (err) {
-        pushError = err;
+        await extensionApi.window.showErrorMessage(
+          `An error occurred while pushing the image '${image.name}:${image.tag}' to Developer Sandbox cluster'${targetSb}'.${pushError}`,
```
Comment (dgolovin) [inline src/extension.ts:163]:
```suggestion
          `An error occurred while pushing the image '${image.name}:${image.tag}' to Developer Sandbox cluster'${targetSb}'.${err}`,
```

**Review by dgolovin — COMMENTED** [review]
Fails for me with
`An error occurred while pushing the image 'docker.io/library/my-custom-image:latest' to Developer Sandbox cluster'sandbox'.Error: (HTTP code 400) unexpected - failed to parse "X-Registry-Auth" header for /images/default-route-openshift-image-registry.apps.sandbox-m3.1530.p1.openshiftapps.com/nivologd-dev/my-custom-image:latest/push: illegal base64 data at input byte 55`

**Comment by dgolovin** [comment]
@jeffmaury Apparently when auth configuration json encoded there is '+' char in base64 string which has index 55 in the string. Registry endpoint complains it is not valid base64 char. After looking around I found that I can use 
```
{
  "identitytoken": "9cbaf023786cd7..."
}
```
config instead.  I tried and push request finishes without error, but nothing actually pushed to the registry.

I verified old pull request which is using podman and it works.

**Comment by jeffmaury** [comment]
That's probably because you registered the Sandbox registry in the registries. But it's not required anymore. But will check. You also need to use the latest versionof the extension API

**Comment by dgolovin** [comment]
> That's probably because you registered the Sandbox registry in the registries. But it's not required anymore.
It seems that dockerode module does not use auth settings. I tried all the combinations, just in case. Did not work for me. It ether did not fail and nothing actually pushed, or failed with error indicates that registry server could not parse base64 encoded authentication info because of 55th char, which is '+' in my case, is not correct base64 encoded char.

I verified if i can encode it when '+' is present in base64 sequence. Works for me. Spec is also saying it is correct char for base64 encoded string.

I am stuck for now.

**Review by dgolovin — COMMENTED** [review]
Still does not work for me. I have the same problem with bad '+' character in base64 encoded authentication configuration.

**Comment by dgolovin** [comment]
@jeffmaury this PR can be merged after latest docker-desktop/api package is published. I implemented workaround to work correctly with openshift internal image registry.

**Comment by jeffmaury** [comment]
So now it is failing for me:

image

**Comment by dgolovin** [comment]
I fixed PR for docker-modem update to 3.0.8 where encoding for X-Registry-Auth is fixed. It when trough without errors https://github.com/containers/podman-desktop/pull/3179.

**Review by dgolovin — APPROVED** [review]

**Pull request merged**